validator/tick/empty_tick.go

```diff
@@ -0,0 +1,67 @@
+package tick
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/qubic/go-archiver/protobuff"
+	"github.com/qubic/go-archiver/store"
+	"github.com/qubic/go-node-connector/types"
+)
+
+func CalculateEmptyTicksForEpoch(ctx context.Context, ps *store.PebbleStore, epoch uint32) (uint32, error) {
+
+	epochs, err := ps.GetProcessedTickIntervals(ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "getting stored epochs")
+	}
+
+	for _, e := range epochs {
+		if e.Epoch != epoch {
+			continue
+		}
+
+		var emptyTicks uint32
+
+		for _, interval := range e.Intervals {
+			for tickOffset := range interval.LastProcessedTick - interval.InitialProcessedTick + 1 {
+				tickNumber := tickOffset + interval.InitialProcessedTick
+
+				tickData, err := ps.GetTickData(ctx, tickNumber)
+				if err != nil {
+					return 0, errors.Wrapf(err, "getting tick data for tick %d", tickNumber)
+				}
+
+				if CheckIfTickIsEmptyProto(tickData) {
+					emptyTicks += 1
+					continue
+				}
+			}
+		}
+		return emptyTicks, err
+	}
+	return 0, nil
+}
+
+func CheckIfTickIsEmptyProto(tickData *protobuff.TickData) bool {
+	if tickData == nil {
+		return true
+	}
+
+	if tickData.VarStruct == nil {
+		return true
+	}
+
+	if tickData.TransactionIds == nil {
```

Comment:
how did you end up with the last 2 if conditions? which are the exact cases when tick data is not nil and the tick is empty?